File: sonar-plsql-plugin/src/main/java/br/com/felipezorzo/sonar/plsql/PlSql.java
package br.com.felipezorzo.sonar.plsql;

import java.util.List;

import org.apache.commons.lang.StringUtils;
import org.sonar.api.config.Settings;
import org.sonar.api.resources.AbstractLanguage;

import com.google.common.collect.Lists;

public class PlSql extends AbstractLanguage {
    
    public static final String KEY = "plsql";
    
    private static final String[] DEFAULT_FILE_SUFFIXES = { "sql", "pkg", "pks", "pkb" };

    private Settings settings;

    public PlSql(Settings configuration) {
      super(KEY, "PL/SQL");
      this.settings = configuration;
    }

    public Settings getSettings() {
      return this.settings;
    }

    @Override
    public String[] getFileSuffixes() {
      String[] suffixes = filterEmptyStrings(settings.getStringArray(PlSqlPlugin.FILE_SUFFIXES_KEY));
      return suffixes.length == 0 ? DEFAULT_FILE_SUFFIXES : suffixes;
    }

    private String[] filterEmptyStrings(String[] stringArray) {
      List<String> nonEmptyStrings = Lists.newArrayList();
      for (String string : stringArray) {
        if (StringUtils.isNotBlank(string.trim())) {
          nonEmptyStrings.add(string.trim());
        }
      }
      return nonEmptyStrings.toArray(new String[nonEmptyStrings.size()]);
    }

}

File: sonar-plsql-open-plugin/src/main/kotlin/org/sonar/plsqlopen/PlSql.kt
/**
 * Z PL/SQL Analyzer
 * Copyright (C) 2015-2019 Felipe Zorzo
 * mailto:felipebzorzo AT gmail DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package org.sonar.plsqlopen

import org.sonar.api.config.Configuration
import org.sonar.api.resources.AbstractLanguage

class PlSql internal constructor(private val settings: Configuration) : AbstractLanguage(KEY, "PL/SQL (ZPA)") {

    private val defaultFileSuffixes = arrayOf("sql", "pkg", "pks", "pkb")

    override fun getFileSuffixes(): Array<String> {
        val suffixes = filterEmptyStrings(settings.getStringArray(PlSqlPlugin.FILE_SUFFIXES_KEY))
        return if (suffixes.isEmpty()) defaultFileSuffixes else suffixes
    }

    private fun filterEmptyStrings(stringArray: Array<String>): Array<String> {
        val nonEmptyStrings = stringArray
            .filterNot { it.isBlank() }
            .map { it.trim() }
        return nonEmptyStrings.toTypedArray()
    }

    companion object {
        const val KEY = "plsqlopen"
    }

}
